src/main/java/org/jabref/logic/externalfiles/LinkedFileHandler.java

```diff
@@ -59,6 +59,10 @@ public boolean moveToDefaultDirectory() throws IOException {
                     filePreferences.getFileDirectoryPattern());
         }
 
+        if (filePreferences.getFileDirectory().isPresent()) {
+            targetDirectoryName = filePreferences.getFileDirectory().get().resolve(targetDirectoryName).toString();
+        }
```

Comment:
Isn't this duplicating the behavior of line 43?

```java
Optional<Path> targetDirectory = databaseContext.getFirstExistingFileDir(filePreferences);
```

If the `FileDirectory` should be used, isn't it already in the content of `targetDirectory`?

Reply:
Yep, the addition does not make any sense

Reply:
And even worse, it overwrites the behavior of the  setting "should store files relative to libary file location"
 overwrites all other directories, so the other tests are actually useless